core/src/utilities/helpers/global-search-helpers.js

```diff
@@ -0,0 +1,15 @@
+/* istanbul ignore file */
+class GlobalSearchHelperClass {
+  constructor() {}
+
+  handleVisibilityGlobalSearch() {
+    const globalSearchCtn = document.querySelector('.lui-global-search');
+    if (globalSearchCtn.offsetWidth <= 384) {
+      globalSearchCtn.classList.add('lui-global-search-toggle');
+    } else {
+      globalSearchCtn.classList.remove('lui-global-search-toggle');
+    }
```

Comment:
```suggestion
    const globalSearchCtn = document.querySelector('.lui-global-search');
    const condition = globalSearchCtn.offsetWidth <= 384;
    globalSearchCtn.classList.toggle('lui-global-search-toggle', condition);
```
`toggle()` seems perfect to be here. FYI: https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle

Please test whether this solution works properly.